util/gen_log_guide.ts

```diff
@@ -0,0 +1,503 @@
+import path from 'path';
+
+import markdownMagic from 'markdown-magic';
+
+import { isLang, languages, NonEnLang } from '../resources/languages';
+import logDefinitions, { LogDefinitionTypes } from '../resources/netlog_defs';
+import NetRegexes from '../resources/netregexes';
+import Regexes from '../resources/regexes';
+import LogRepository from '../ui/raidboss/emulator/data/network_log_converter/LogRepository';
+import ParseLine from '../ui/raidboss/emulator/data/network_log_converter/ParseLine';
+
+type LangObject<T> = { en: T } & { [lang in NonEnLang]?: T };
+
+const curPath = path.resolve();
+
+// Exclude these types since they're not relevant or covered elsewhere
+type ExcludedLineDocs =
+  | 'None'
+  | 'addBuff'
+  | 'removeBuff'
+  | 'flyingText'
+  | 'outgoingAbility'
+  | 'incomingAbility'
+  | 'combatantHP'
+  | 'networkAOEAbility'
+  | 'networkWorld'
+  | 'networkEffectResult'
+  | 'parserInfo'
+  | 'processInfo'
+  | 'debug'
+  | 'packetDump'
+  | 'version'
+  | 'error'
+  | 'timer';
+
+// @TODO: These types are missing or unused
+type MissingLineDocs = 'Map';
+
+type LineDocTypes = Exclude<LogDefinitionTypes, ExcludedLineDocs | MissingLineDocs>;
+
+type LineDocType = {
+  // We can generate `network` type automatically for everything but regex
+  regexes?: {
+    network: string;
+    logLine: string;
+  };
+  examples: LangObject<readonly string[]>;
+};
+
+type LineDocs = {
+  [type in LineDocTypes]: LineDocType;
+};
+
+// @TODO: Remove Partial once everything is mapped
+const lineDocs: LineDocs = {
+  GameLog: {
+    regexes: {
+      network: NetRegexes.gameLog({ capture: true }).source,
```

Comment:
What do you think about just having `regex: 'gameLog'` here, and not repeating `{ capture: true }).source` everywhere?

Reply:
I thought about that originally but decided against it so that regexes could be manually specified if needed. There's also the case of `nameToggle` where the key exists on one but not the other.